### PR TITLE
Appendix on protocol rationale

### PR DESCRIPTION
**Pull request by fimbault**
Fixes issue #176

Note : in the appendix, I used "OAuth 2", elsewhere we talk about "OAuth 2.0". Maybe needs some adjustements.

### CODE REVIEW

File: draft-ietf-gnap-core-protocol.md [hunk 1]
```diff
@@ -4559,6 +4560,7 @@ sure that it has the permission to do so.
     - Changed "key_proofs" to "key_proofs_supported".
     - Changed "assertions" to "assertions_supported".
     - Updated discovery and field names for subject formats.
+    - Add an appending to provide protocol rationale, compared to OAuth2.
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:4563]:
```suggestion
    - Add an appendix to provide protocol rationale, compared to OAuth2.
```

**Review by jricher — APPROVED** [review]
Looks good, just one typo to fix before inclusion.

File: draft-ietf-gnap-core-protocol.md (hunk 1 above)
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md]: Thanks for spotting this!

File: draft-ietf-gnap-core-protocol.md [hunk 2]
```diff
@@ -4593,6 +4595,33 @@ sure that it has the permission to do so.
...
+
+4. **Expanded delegation:**
+OAuth 2 defines the “scope” parameter for controlling access to APIs. This parameter has been coopted to mean a number of different things in different protocols, including flags for turning special behavior on and off, including the return of data apart from the access token. The “resource” parameter and RAR extensions expand on the “scope” concept in similar but different ways.
+GNAP defines a rich structure for requesting access, with string references as an optimization. GNAP defines methods for requesting directly-returned user information, separate from API access. This information includes identifiers for the current user and structured assertions. The core GNAP protocol makes no assumptions or demands on the format or contents of the access token, but the RS extension allows a negociation of token formats between the AS and RS.
```
Comment (yaronf) [inline draft-ietf-gnap-core-protocol.md:4615]: negociation - typo

File: draft-ietf-gnap-core-protocol.md [hunk 3]
```diff
@@ -4593,6 +4595,33 @@ sure that it has the permission to do so.
...
+
+3. **Client instances:**
+OAuth 2 requires all clients to be registered at the AS and to use a client_id known to the AS as part of the protocol. This client_id is generally assumed to be assigned by a trusted authority during a registration process, and OAuth places a lot of trust on the client_id as a result. Dynamic registration allows different classes of clients to get a client_id at runtime, even if they only ever use it for one request.
+GNAP allows the client instance to present an unknown key to the AS and use that key to protect the ongoing request. GNAP’s client instance identifier mechanism allows for pre-registered clients and dynamically registered clients to exist as an optimized case without requiring the identifier as part of the protocol at all times.
+
+4. **Expanded delegation:**
+OAuth 2 defines the “scope” parameter for controlling access to APIs. This parameter has been coopted to mean a number of different things in different protocols, including flags for turning special behavior on and off, including the return of data apart from the access token. The “resource” parameter and RAR extensions expand on the “scope” concept in similar but different ways.
```
Comment (yaronf) [inline draft-ietf-gnap-core-protocol.md:4614]: RAR - ref to draft-ietf-oauth-rar

File: draft-ietf-gnap-core-protocol.md [hunk 4]
```diff
@@ -4593,6 +4595,33 @@ sure that it has the permission to do so.
...
+5. **Cryptography-based security:**
+OAuth 2 uses shared bearer secrets, including the client_secret and access token, and advanced authentication and sender constraint have been built on after the fact in inconsistent ways.
+In GNAP, all communication between the client instance and AS is bound to a key held by the client instance. GNAP uses the same cryptographic mechanisms for both authenticating the client (to the AS) and binding the access token (to the RS and the AS). GNAP allows extensions to define new cryptographic protection mechanisms, as new methods are expected to become available over time. GNAP does not have a notion of “public clients” because key information can always be sent and used dynamically.
+
+6. **Privacy and usable security:** 
+OAuth 2's deployment model assumes a strong binding between the AS and the RS.
+GNAP is designed to be interoperable with decentralized identity standards and to provide a human-centric authorization layer. In addition to the core protocol, GNAP provides an extension that supports various patterns of communication between RSs and ASs. GNAP tries to limit the odds of a consolidation to just a handful of super-popular AS services.
```
Comment (yaronf) [inline draft-ietf-gnap-core-protocol.md:4623]: "provides an extension" - an extension mechanism? An extension point? Extensibility?

**Review by yaronf — CHANGES_REQUESTED** [review]

File: draft-ietf-gnap-core-protocol.md [hunk 5]
```diff
@@ -106,7 +106,8 @@ OpenID Connect {{OIDC}}, and the family of protocols that have grown up
 around that ecosystem. However, GNAP is not an extension of OAuth 2.0
 and is not intended to be directly compatible with OAuth 2.0. GNAP seeks to
 provide functionality and solve use cases that OAuth 2.0 cannot easily
-or cleanly address. Even so, GNAP and OAuth 2.0 will exist in parallel 
+or cleanly address. {{vs-oauth2}} further details the protocol rationale compared to OAuth 2.0.
+Even so, GNAP and OAuth 2.0 will exist in parallel
```
[the remark this answers is not in the document]
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md:110]: yes

File: draft-ietf-gnap-core-protocol.md [hunk 6]
```diff
@@ -4593,6 +4595,33 @@ sure that it has the permission to do so.
...
+OAuth 2 generally assumes the user has access to the a web browser. The type of interaction available is fixed by the grant type, and the most common interactive grant types start in the browser. OAuth 2 assumes that the user using the client software is the same user that will interact with the AS to approve access.
+GNAP allows various patterns to manage authorizations and consents required to fulfill this requested delegation, including information sent by the client instance, information supplied by external parties, and information gathered through the interaction process. GNAP allows a client instance to list different ways that it can start and finish an interaction, and these can be mixed together as needed for different use cases. GNAP interactions can use a browser, but don’t have to. Methods can use inter-application messaging protocols, out-of-band data transfer, or anything else. GNAP allows extensions to define new ways to start and finish an interaction, as new methods and platforms are expected to become available over time. GNAP is designed to allow these users to be two different people, but still works in the optimized case of them being the same party.
```
[the remark this answers is not in the document]
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md:4603]: yes. the end-user and the resource owner

File: draft-ietf-gnap-core-protocol.md (hunk 2 above)
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md:4615]: yes

File: draft-ietf-gnap-core-protocol.md (hunk 3 above)
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md:4614]: yes

File: draft-ietf-gnap-core-protocol.md (hunk 4 above)
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md:4623]: at this stage, extensibility will do

**Pull request merged**